packages/thirdweb/src/utils/any-evm/deploy-metadata.ts

```diff
@@ -50,6 +50,7 @@ export async function fetchDeployMetadata(
     ...parsedMeta,
     version: rawMeta.version,
     bytecode: deployBytecode,
+    name: rawMeta.name,
```

Comment:
can you explain this fix ?

Reply:
Both `rawMeta` and `parsedMeta` have the field `name`.
But rawMeta.name is actually the contract-id used for publishing, whereas parsedMeta.name is the target contract name.

For handling custom id, we need rawMeta.name, but currently it gets replaced with parsedMeta.name due to the order we deconstruct these objects.

So, either we reverse the order, or reassign name as `rawMeta.name`

Ref from legacy package: https://github.com/thirdweb-dev/js/pull/3424/files#diff-a8673381af2cf9fa6622981b06dd878c1e9e76b6f8d7af5017b2eec29ac4933f

Reply:
makes sense